gui/src/pages/gui/ToolCallDiv/ToolTruncateHistoryIcon.tsx

```diff
@@ -0,0 +1,40 @@
+import { BarsArrowUpIcon } from "@heroicons/react/24/outline";
+import { chatMessageIsEmpty } from "core/llm/messages";
+import { findLastIndex } from "core/util/findLast";
+import { useMemo } from "react";
+import { useMainEditor } from "../../../components/mainInput/TipTapEditor";
+import { ToolbarButtonWithTooltip } from "../../../components/StyledMarkdownPreview/StepContainerPreToolbar/ToolbarButtonWithTooltip";
+import { useAppDispatch, useAppSelector } from "../../../redux/hooks";
+import { truncateHistoryToMessage } from "../../../redux/slices/sessionSlice";
+
+export function ToolTruncateHistoryIcon({
+  historyIndex,
+}: {
+  historyIndex: number;
+}) {
+  const isStreaming = useAppSelector((state) => state.session.isStreaming);
+  const history = useAppSelector((state) => state.session.history);
+  const lastMessageIndex = useMemo(() => {
+    return findLastIndex(history, (item) => !chatMessageIsEmpty(item.message));
+  }, [history]);
+  const dispatch = useAppDispatch();
+  const { mainEditor } = useMainEditor();
+  if (isStreaming || historyIndex === lastMessageIndex) {
+    return null;
+  }
+  return (
+    <ToolbarButtonWithTooltip
+      tooltipContent="Truncate chat to this message"
```

Comment:
```suggestion
      tooltipContent="Trim chat to this message"
```

"Truncate" feels a bit jargon-y to me